src/transformers/models/fastspeech2/modeling_fastspeech2.py

```diff
@@ -0,0 +1,1199 @@
+# coding=utf-8
+# Copyright 2021 The Ontocord team, the G2P, Melgan, and Fastspeech2 Authors, and the HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This software is based on other open source code. A huge thanks to
+# the Huggingface team, and the authors of the Fastspeech2 and Melgan
+# papers and the following authors who originally implemented the
+# various modules, from which this code is based:
+
+# Chung-Ming Chien's Fastspeech2 implementation is under the MIT license: https://github.com/ming024/FastSpeech2
+# Seung-won Park 박승원's Meglan implementation is under BSD-3 license: https://github.com/seungwonpark/melgan
+# Kyubyong Park's G2P implementation is under the Apache 2 license: https://github.com/Kyubyong/g2p, and also here for pytorch specifics https://github.com/Kyubyong/nlp_made_easy/blob/master/PyTorch%20seq2seq%20template%20based%20on%20the%20g2p%20task.ipynb
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import numpy as np
+from collections import OrderedDict
+import copy
+import math
+from .configuration_fastspeech2 import FastSpeech2Config
+from transformers import PreTrainedModel
+from torch.nn.utils.rnn import pack_padded_sequence, pad_packed_sequence
+import warnings
+
+### G2P
+# A simple Seq2Seq GRU->GRU network to fix and guess phoenemes from
+# graphenes. Tie the embeddings to the fastspeech2 embeddings to
+# increase generalizations.
+
+class G2PEncoder(nn.Module):
```

Comment:
I'm a bit surprised to find the `nn.Module` here.

I think FastSpeech2 does not contain a `nn.GRU` layer no? 
Neither the [paper](https://arxiv.org/pdf/2006.04558.pdf), nor the other [implementation](https://github.com/ming024/FastSpeech2) mention anything of a gated recurrent unit. Could you elaborate a bit here? :-)

Reply:
Ah sorry, just read your PR description. I think it will be difficult to merge user-specific code to `src/transformers` for us -> do you think there could be a possibility to merge the official `FastSpeech2` implementation and then maybe just add a script under `examples/research_projects/fast_speech_2` that includes costum modules?

Reply:
No worries @patrickvonplaten. I'm happy to help you guys merge the officeial FastSpeech2 if someone can take a crack at it first, I can contribute :)